optuna/storages/journal/_file.py

```diff
@@ -128,26 +129,52 @@ class JournalFileSymlinkLock(BaseJournalFileLock):
     Args:
         filepath:
             The path of the file whose race condition must be protected.
+        grace_period:
+            Grace period before an existing lock is forcibly released.
     """
 
-    def __init__(self, filepath: str) -> None:
+    def __init__(self, filepath: str, grace_period: int = 30) -> None:
         self._lock_target_file = filepath
         self._lock_file = filepath + LOCK_FILE_SUFFIX
+        if grace_period <= 0:
+            raise ValueError("The value of `grace_period` should be a positive integer.")
+        if grace_period < 3:
+            warnings.warn("The value of `grace_period` might be too small. ")
+        self.grace_period = grace_period
 
     def acquire(self) -> bool:
         """Acquire a lock in a blocking way by creating a symbolic link of a file.
 
         Returns:
             :obj:`True` if it succeeded in creating a symbolic link of ``self._lock_target_file``.
-
         """
         sleep_secs = 0.001
+        last_update_time = time.monotonic()
+        mtime = None
         while True:
             try:
                 os.symlink(self._lock_target_file, self._lock_file)
                 return True
             except OSError as err:
                 if err.errno == errno.EEXIST:
+                    try:
+                        current_mtime = os.stat(self._lock_file).st_mtime
+                    except OSError:
+                        continue
+                    if current_mtime != mtime:
+                        mtime = current_mtime
+                        last_update_time = time.monotonic()
+
+                    if time.monotonic() - last_update_time > self.grace_period:
+                        warnings.warn(
+                            "The existing lock file has not been released "
+                            "for an extended period. Forcibly releasing the lock file."
+                        )
+                        try:
+                            self.release()
```

Comment:
Would it be a good idea to reset `sleep_secs` to `0.001` after calling `release()`? Another thread may acquire the lock between your `release()` and `os.symlink()`.